infoschema/tables.go

```diff
@@ -1909,25 +1908,51 @@ func dataForServersInfo() ([][]types.Datum, error) {
 	return rows, nil
 }
 
-func dataForTiDBClusterInfo(ctx sessionctx.Context) ([][]types.Datum, error) {
+// ServerInfo represents the basic server information of single cluster component
+type ServerInfo struct {
+	ServerType string
+	Address    string
+	StatusAddr string
+	Version    string
+	GitHash    string
+}
+
+// GetClusterServerInfo returns all components information of cluster
+func GetClusterServerInfo(ctx sessionctx.Context) ([]ServerInfo, error) {
```

Comment:
No need export this function?

Reply:
The` perfschema` will reuse this function.